benchmark/NCM_Stress_Test_Report/NCM Stress Test Report.adoc

```diff
@@ -0,0 +1,97 @@
+= NCM Stress Test Report
+:revnumber: v1.0
+:revdate: 2022 January 28
+:author: Rio Zhu
+:email: zzhu@futurewei.com
+
+:toc: right
+:imagesdir: images
+
+== Overview
+
+This document presents the results of running on-demand stress tests against Network-Configuration-Manager(NCM), with Ignite as the cache layer of NCM. The reason of this test is to get a better understanding of how much on-demand requests can NCM handle in a certain amount of time.
+
+The following graph shows the setup and the workflow of this test.
+
+We tested NCM with different numbers of VPCs, and different number of ports inside each VPC.
+
+image::ncm_stress_test_setup.png[test_setup, 800]
+
+== Test Setup and Detailed Workflow
+
+The test setup is very similar with our previous tests on the on-demand workflow. Except that we replced one of the Alcor-Control-Agent(ACA) on one of the nodes with the newly written Dubhe Agent, a prototype written in Golang that mimics the gRPC server/client ability of ACA. Of course, for the Test Controller, changes were made in order to support the multi-vpc test scenarios.
+
+At the beginning of the test, the Test Controller first generates the GoalStateV2 messages with the number of VPCs, and number of ports inside each VPC, and it sends those GoalStateV2s to NCM.
+
+NCM, after receiving and storing these GoalStateV2s into its cache, sends down the GoalStates to the host agents, which are the Dubhe agent on Node 1 and ACA on Node 2. 
```

Comment:
That's a good question. Whatever agent on the Node 2 doesn't really matter in these tests, it could be the Dubhe Agent or ACA. That agent's role is only to be there and receive the GoalStateV2 that was created by the TC. But in the future we can definitely replace this ACA with the Dubhe agent and then find out a way to let these agents to send the on-demand requests at the same time. I will add this to the `Future Work` section.